clients/client-python/build.gradle.kts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+import io.github.piyushroshan.python.VenvTask
+
+plugins {
+  java
+  id("io.github.piyushroshan.python-gradle-miniforge-plugin") version "1.0.0"
+}
+
+pythonPlugin {
+  pythonVersion.set(project.rootProject.extra["pythonVersion"].toString())
+}
+
+fun deleteCacheDir(targetDir: String) {
+  project.fileTree(project.projectDir).matching {
+    include("**/$targetDir/**")
+  }.forEach { file ->
+    val targetDirPath = file.path.substring(0, file.path.lastIndexOf(targetDir) + targetDir.length)
+    project.file(targetDirPath).deleteRecursively()
+  }
+}
+
+tasks {
+  register<VenvTask>("condaInfo") {
+    venvExec = "conda"
+    args = listOf("info")
+  }
+
+  val pipInstall by registering(VenvTask::class) {
+    venvExec = "pip"
+    args = listOf("install", "--isolated", "-r", "requirements.txt")
+  }
+
+  val runPyTests by registering(VenvTask::class) {
+    dependsOn(pipInstall)
+    venvExec = "pytest"
+    workingDir = projectDir.resolve("tests")
+  }
+
+  compileJava {
+    dependsOn(pipInstall)
```

Comment:
should the `compile` and `spotless` depend on `pip install`?  Python doesn't need to compile and can't verify the correctness of the program. If the user only updates the Java code, they also need to download the dependencies of the Python module. How about just making the `test` depend on `pip install`?

Reply:
Yes, Currently use Gradle to comple Python module have this limit (need add `compileJava`).
I Create an issue #2770 to track this probleam.

Reply:
hi @coolderli I fixed this problem in this PR. :-)